### PR TITLE
Added nearly fully syntatic support for MATCH

### PR DESCRIPTION
**Pull request by dutor**
Exceptions are:
 * `CALL`
 * Mutiple labels on one vertex.

### CODE REVIEW

**Review by CPWstatic — DISMISSED** [review]

File: .linters/cpp/checkKeyword.py [hunk 1]
```diff
@@ -97,16 +97,13 @@
     'KW_PROP',
     'KW_BALANCE',
     'KW_STOP',
-    'KW_LIMIT',
```
Comment (laura-ding) [inline .linters/cpp/checkKeyword.py:100]: why make the limit and offset to Unreserved keyword

File: src/parser/parser.yy [hunk 2]
```diff
@@ -1023,29 +1118,100 @@ match_edge_prop
     : %empty {
         $$ = nullptr;
     }
-    | L_BRACKET match_alias match_edge_type R_BRACKET {
-        $$ = new MatchEdgeProp($2, $3, nullptr);
+    | L_BRACKET match_alias opt_match_edge_type_list match_step_range R_BRACKET {
+        UNUSED($4);
```
Comment (laura-ding) [inline src/parser/parser.yy:1122]: maybe need to delete

File: src/parser/parser.yy [hunk 3]
```diff
@@ -1023,29 +1118,100 @@ match_edge_prop
...
+        $$ = new MatchEdgeProp($2, $3, $4, nullptr);
     }
-    | L_BRACKET match_alias match_edge_type map_expression R_BRACKET {
-        $$ = new MatchEdgeProp($2, $3, $4);
+    | L_BRACKET match_alias opt_match_edge_type_list match_step_range map_expression R_BRACKET {
+        UNUSED($4);
```
Comment (laura-ding) [inline src/parser/parser.yy:1126]: ditto

File: src/parser/test/ParserTest.cpp [hunk 4]
```diff
@@ -2338,6 +2338,180 @@ TEST(Parser, Match) {
...
+        std::string query = "UNWIND a AS b MATCH (c) -[m]- (d) RETURN b,c,d";
+        auto result = parser.parse(query);
+        ASSERT_TRUE(result.ok()) << result.status();
+    }
+    {
+        GQLParser parser;
+        std::string query = "UNWIND a AS b MATCH (c) MATCH (d) WITH e MATCH (f) RETURN b,c,d";
+        auto result = parser.parse(query);
+        ASSERT_TRUE(result.ok()) << result.status();
+    }
```
Comment (CPWstatic) [inline src/parser/test/ParserTest.cpp:2514]:
```
MATCH p = (c) -[m]- (d) RETURN p
```

File: src/parser/parser.yy (hunk 2 above)
Comment (dutor) [reply src/parser/parser.yy:1122]: thanks!

File: .linters/cpp/checkKeyword.py (hunk 1 above)
Comment (dutor) [reply .linters/cpp/checkKeyword.py:100]: I think we should be less aggressive on making a keyword reserved.

File: src/parser/scanner.lex [hunk 5]
```diff
@@ -485,16 +491,7 @@ FORMAT                      ([Ff][Oo][Rr][Mm][Aa][Tt])
...
-                                try {
-                                    folly::StringPiece text(yytext, yyleng);
-                                    yylval->doubleval = folly::to<double>(text);
-                                } catch (...) {
-                                    throw GraphParser::syntax_error(*yylloc, "Out of range:");
-                                }
-                                return TokenType::DOUBLE;
-                            }
-{DEC}*\.{DEC}+              {
+{DEC}+\.{DEC}+              {
```
Comment (Shylock-Hg) [inline src/parser/scanner.lex:494]: Why remove this?

Comment (dutor) [reply src/parser/scanner.lex:494]: It conflicts with a new added syntax `1..2`.
But I think we could do some tricks to deal with this, maybe later on.

**Review by laura-ding — APPROVED** [review]

**Review by CPWstatic — APPROVED** [review]

**Pull request merged**